front/lib/api/assistant/agent_usage.ts

```diff
@@ -146,67 +147,48 @@ export async function agentMentionsCount(
   agentConfiguration?: LightAgentConfigurationType,
   rankingUsageDays: number = RANKING_USAGE_DAYS
 ): Promise<AgentUsageCount[]> {
-  // We retrieve mentions from conversations in order to optimize the query
-  // Since we need to filter out by workspace id, retrieving mentions first
-  // would lead to retrieve every single messages
-  const mentions = await Conversation.findAll({
-    attributes: [
-      [
-        Sequelize.literal('"messages->mentions"."agentConfigurationId"'),
-        "agentConfigurationId",
-      ],
-      [
-        Sequelize.fn("COUNT", Sequelize.literal('"messages->mentions"."id"')),
-        "messageCount",
-      ],
-      [
-        Sequelize.fn("COUNT", Sequelize.literal("DISTINCT conversation.id")),
-        "conversationCount",
-      ],
-      [
-        Sequelize.fn(
-          "COUNT",
-          Sequelize.literal('DISTINCT "messages->userMessage"."userId"')
-        ),
-        "userCount",
-      ],
-    ],
-    where: {
-      workspaceId,
-    },
-    include: [
-      {
-        model: Message,
-        required: true,
-        attributes: [],
-        include: [
-          {
-            model: Mention,
-            as: "mentions",
-            required: true,
-            attributes: [],
-            where: {
-              ...(agentConfiguration
-                ? { agentConfigurationId: agentConfiguration.sId }
-                : {}),
-              createdAt: {
-                [Op.gt]: literal(`NOW() - INTERVAL '${rankingUsageDays} days'`),
-              },
-            },
-          },
-          {
-            model: UserMessage,
-            as: "userMessage",
-            required: true,
-            attributes: [],
-          },
-        ],
+  const readReplica = getFrontReplicaDbConnection();
+
+  if (typeof rankingUsageDays !== "number") {
+    // Prevent SQL injection
+    throw new Error("Invalid ranking usage days");
+  }
+
+  const mentions = await readReplica.query(
+    `
+    WITH message_counts AS (
+      SELECT 
+        mentions."agentConfigurationId",
+        COUNT(DISTINCT mentions.id) as message_count,
+        COUNT(DISTINCT c.id) as conversation_count, 
+        COUNT(DISTINCT um."userId") as user_count
+      FROM conversations c
+      INNER JOIN messages m ON m."conversationId" = c.id 
+      INNER JOIN mentions ON mentions."messageId" = m.id
+      INNER JOIN user_messages um ON um.id = m."userMessageId"
+      WHERE 
+        c."workspaceId" = :workspaceId
+        AND mentions."workspaceId" = :workspaceId
+        AND mentions."createdAt" > NOW() - INTERVAL '${rankingUsageDays} days'
```

Comment:
`nit`: let's use replacement for `rankingUsageDays` too?

Reply:
sequelize being sequelize